logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java

```diff
@@ -0,0 +1,872 @@
+                }
+                parseFields(context.getPropertyValue(FIELDS).asString());
+                checkTagsAndFieldsForDuplicates();
+                break;
+            case ALL_AS_FIELDS:
+                if (context.getPropertyValue(TAGS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + TAGS.getName() + " to not be set.");
+                }
+                if (context.getPropertyValue(FIELDS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + FIELDS.getName() + " to not be set.");
+                }
+                break;
+            case ALL_AS_TAGS_BUT_EXPLICIT_FIELDS:
+                if (context.getPropertyValue(TAGS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + TAGS.getName() + " to not be set.");
+                }
+                if (!context.getPropertyValue(FIELDS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + FIELDS.getName() + " to be set.");
+                }
+                parseFields(context.getPropertyValue(FIELDS).asString());
+                break;
+            case ALL_AS_FIELDS_BUT_EXPLICIT_TAGS:
+                if (!context.getPropertyValue(TAGS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + TAGS.getName() + " to be set.");
+                }
+                parseTags(context.getPropertyValue(TAGS).asString());
+                if (context.getPropertyValue(FIELDS).isSet())
+                {
+                    throw new InitializationException("Configuration mode " + mode + " requires " + FIELDS.getName() + " to not be set.");
+                }
+                break;
+        }
+
+        // Parse time fields if any
+        if (context.getPropertyValue(TIME_FIELD).isSet())
+        {
+            parseTimeFields(context.getPropertyValue(TIME_FIELD).asString());
+            checkTagsOrFieldsTimeFieldCollision();
+        }
+
+        // Consistency level
+        if (context.getPropertyValue(CONSISTENCY_LEVEL).isSet())
+        {
+            consistencyLevel = ConsistencyLevel.valueOf(context.getPropertyValue(CONSISTENCY_LEVEL).asString());
+        }
+
+        // Retention policy
+        if (context.getPropertyValue(RETENTION_POLICY).isSet())
+        {
+            retentionPolicy = context.getPropertyValue(RETENTION_POLICY).asString();
+        }
+
+        // Url
+        String influxUrl = context.getPropertyValue(URL).asString();
+
+        // Sanity checking
+        if (context.getPropertyValue(PASSWORD).isSet() && !context.getPropertyValue(USER).isSet())
```

Comment:
This should be done in method :
protected Collection<ValidationResult> customValidate(ValidationContext context)
instead

Reply:
Done